Update the sdk target to net11.0

DonnaChen888 opened the pull request.

DonnaChen888 committed:
Update the sdk target to net11.0

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

DonnaChen888 committed:
Update WindowsDesktopTools to net10.0

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

ViktorHofer commented on eng/restore-toolset.ps1:
```diff
@@ -19,6 +19,7 @@ function InitializeCustomSDKToolset {
   InstallDotNetSharedFramework "7.0.0"
   InstallDotNetSharedFramework "8.0.0"
   InstallDotNetSharedFramework "9.0.0"
+  InstallDotNetSharedFramework "10.0.0-rtm.25523.111"
```
Shouldn't this be just 10.0.0?

marcpopMSFT committed:
Fix exit code handling in restore-toolset.ps1

The install-scripts won't always set last exit code like if the runtime is already installed so pre-set it.

marcpopMSFT committed:
The tests are complaining as the max runtime is still 10 because we don't have an 11 runtime yet. There's not a good central place to set this as different tests use different DBPs so try setting this in the run tests script

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

marcpopMSFT committed:
Update currenttargetframework to 11

marcpopMSFT committed:
Update the restore toolset to target 10.0.0

marcpopMSFT committed:
Update all of the changes to source to use the SDK Target Framework

marcpopMSFT committed:
Update the maximum version and set the default framework back

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

ViktorHofer commented on src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.5_0.targets:
```diff
@@ -27,10 +27,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     <!-- Paths to tools, tasks, and extensions are calculated relative to the BlazorWebAssemblySdkDirectoryRoot. This can be modified to test a local build. -->
     <BlazorWebAssemblySdkDirectoryRoot Condition="'$(BlazorWebAssemblySdkDirectoryRoot)'==''">$(MSBuildThisFileDirectory)..\</BlazorWebAssemblySdkDirectoryRoot>
-    <_BlazorWebAssemblySdkTasksTFM Condition=" '$(MSBuildRuntimeType)' == 'Core'">net11.0</_BlazorWebAssemblySdkTasksTFM>
+    <_BlazorWebAssemblySdkTasksTFM Condition=" '$(MSBuildRuntimeType)' == 'Core'">$(SdkTargetFramework)</_BlazorWebAssemblySdkTasksTFM>
```
@marcpopMSFT this is a shipping file in the SDK and can't use this repo's infrastructure:

> C:\Program Files\dotnet\sdk\10.0.100\Sdks\Microsoft.NET.Sdk.BlazorWebAssembly\targets\Microsoft.NET.Sdk.BlazorWebAssembly.5_0.targets

marcpopMSFT replied:
We have a NETCoreAppMaximumVersion property that pulls from the _NETCoreAppPackageVersion so it would match whatever runtime was included in the SDK. Would that work here then?

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0

DonnaChen888 committed:
Merge branch 'main' into dev/Donna/Update-the-sdk-target-to-net11.0
